R/source.R

```diff
@@ -429,33 +434,30 @@ check_cargo_output <- function(compilation_result, message_buffer, tty_has_color
     jsonlite::parse_json
   )
 
-  if (!isTRUE(quiet)) {
-    purrr::walk(
-      gather_cargo_output(
-        cargo_output,
-        "warning",
-        tty_has_colors
-      ),
-      ui_w
-    )
-  }
-
   if (!isTRUE(compilation_result$status == 0)) {
-    error_messages <- purrr::map_chr(
-      gather_cargo_output(
-        cargo_output,
-        "error",
-        tty_has_colors
-      ),
-      bullet_x
+    error_messages <- gather_cargo_output(
+      cargo_output,
+      "error",
+      tty_has_colors
     )
 
-    ui_throw(
-      "Rust code could not be compiled successfully. Aborting.",
+    errors_formatted <- purrr::map_chr(
       error_messages,
+      cli::format_inline,
+      keep_whitespace = TRUE
+    ) %>%
+      # removing double new lines with single new line
+      stringi::stri_replace_all_fixed("\n\n", "\n") %>%
+      # ensures that the leading cli style `x` is there
+      rlang::set_names("x")
+
+    rlang::abort(
+      c(
+        "Rust code could not be compiled successfully. Aborting.",
+        unlist(errors_formatted)
```

Comment:
Why would `errors_formatted` be a list here?

Reply:
You're right. It no longer is. This can be removed. Previously was setting the names in the purrr call which required the output to be a list as `_chr` suffix does not honor the use of named vectors.